airbyte-integrations/connectors/source-surveycto/source_surveycto/source.py

```diff
@@ -109,9 +109,36 @@ def read_records(self, *args, **kwargs) -> Iterable[Mapping[str, Any]]:
 
 # Source
 class SourceSurveycto(AbstractSource):
-    def check_connection(self, logger, config) -> Tuple[bool, any]:
-        return True, None
 
+    
+    def check_connection(self, logger, config) -> Tuple[bool, Any]:
+        
+        server_name = config['server_name']
+        form_id = config["form_id"][0]
+        username = config['username']
+        password = config['password']
+        
+        if not all([server_name, form_id, username, password]):
+            return False, "Incomplete configuration. Please provide 'server_name', 'form_id', 'username', and 'password'."
```

Comment:
This is not necessary because the spec has the fields as required but it doesn't cause any problem too.